### PR TITLE
SDN-4101: Remove HyperShift API dependency, Bump kubernetes dependencies

### PR DESCRIPTION
**Pull request by kyrtapz**
This PR removes the HyperShift API dependency and bumps kubernetes deps to v0.28.4.
Once we remove the SBDB route in HyperShift we could clean up the code by removing the `Services` field from `HostedControlPlane`.

### CODE REVIEW

**Comment by kyrtapz** [comment]
CI is out: https://redhat-internal.slack.com/archives/CBN38N3MW/p1700750291680499

File: pkg/controller/statusmanager/status_manager.go [hunk 1]
```diff
@@ -287,20 +287,34 @@ func (status *StatusManager) writeHypershiftStatus(operStatus *operv1.NetworkSta
...
-		oldStatus := hcp.Status.DeepCopy()
+		var conditions []metav1.Condition
+		jsonData, err := json.Marshal(conditionsRaw)
+		if err != nil {
+			return fmt.Errorf("error marshalling JSON: %v\n", err)
+		}
+		err = json.Unmarshal(jsonData, &conditions)
+		if err != nil {
+			return fmt.Errorf("Error unmarshalling JSON: %v\n", err)
+		}
```
Comment (jcaamano) [inline pkg/controller/statusmanager/status_manager.go:311]: This is also candidate to be moved to the hypershift package. What I am thinking move all ugly unstructured code for hypershift there. But just a nit.

File: pkg/controller/statusmanager/status_manager.go [hunk 2]
```diff
@@ -287,20 +287,34 @@ func (status *StatusManager) writeHypershiftStatus(operStatus *operv1.NetworkSta
...
+
+		oldConditions := make([]metav1.Condition, len(conditions))
```
Comment (jcaamano) [inline pkg/controller/statusmanager/status_manager.go:313]: This whole condition handling could also me moved to the hypershift package, soemhting like maybe `AddConditiontoHostedControlPlane`?

**Pull request merged**